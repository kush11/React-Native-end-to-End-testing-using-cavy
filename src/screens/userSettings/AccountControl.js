import React, { Component } from 'react';
import {
  View, StyleSheet, Text, Alert, AsyncStorage,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import RazorpayCheckout from 'react-native-razorpay';
import { LoginManager, AccessToken } from 'react-native-fbsdk';
import { GoogleSignin } from 'react-native-google-signin';
import { connect } from 'react-redux';
import {
  Left, Body, Right, List, ListItem, Toast
} from 'native-base';
import { fontMaker } from '../../components/utility/fonts/FontMaker';
import {
  updateUsername, updateDob, updateUserSocialImage, updateFetchedUrl, resetState
} from '../../store/actions/index';
import { updateCurrentFlow } from '../../store/actions/assessment';
import { BASE_URL, headers } from '../../api/config/Config';


const mapStateToProps = state => ({
  uName: state.User.name,
  uSocialImage: state.User.socialImage,
  fetchedUrl: state.User.fetchedURL,
  gender: state.User.gender
});
const mapDispatchToProps = dispatch => ({

  updatedUserName: name => dispatch(updateUsername(name)),
  updatedFetchedUrl: url => dispatch(updateFetchedUrl(url)),
  updatedUserSocialImage: uri => dispatch(updateUserSocialImage(uri)),
  updatedDob: dob => dispatch(updateDob(dob)),
  updatedCurrentFlow: flow => dispatch(updateCurrentFlow(flow)),
  ResetState: () => dispatch(resetState())
});

export class AccountSetting extends Component {
  logout = async () => {
    const { navigation, ResetState } = this.props;
    try {
      await ResetState();
    } catch (error) {
      Toast.show({
        text: 'Error Occured',
        duration: 2000,
        type: 'default'
      });
    }

    this.signOut();
    const { updatedCurrentFlow } = this.props;
    navigation.navigate('StartPage');
    updatedCurrentFlow('NEW');
  }

  signOut = async () => {
    const { updatedDob } = this.props;
    updatedDob(null);
    try {
      await AsyncStorage.removeItem('userToken');
      await AsyncStorage.removeItem('UserSocialImage');
      if (await GoogleSignin.isSignedIn()) { await GoogleSignin.signOut(); }
    } catch (error) {
      Toast.show({
        text: 'Error Occured',
        duration: 2000,
        type: 'default'
      });
    }
    if (AccessToken.getCurrentAccessToken()) {
      await LoginManager.logOut();
    }
  };

  showAlert = () => {
    Alert.alert(
      'Account Deactivated',
      'Your account will be deleted within 50 days. You can login within this period with your credentials.',
      [
        {
          text: 'Ok',
          onPress: () => {
            this.logout();
          }
        },
      ],
      { cancelable: false }
    );
  }

  deleteAccount = () => {
    const { uName } = this.props;
    Alert.alert(
      'Delete Account',
      'Do you want to delete your Account?',
      [
        {
          text: 'NO',
          onPress: () => { },
          style: 'cancel'
        },
        {
          text: 'YES',
          onPress: () => {
            AsyncStorage.removeItem('enableTouchId');
            fetch(`${BASE_URL}/api/updateUserStatus`, {
              method: 'POST',
              headers,
              body: JSON.stringify({ name: uName })
            }).then(() => {
              this.showAlert();
            });
          }
        },
      ],
      { cancelable: false }
    );
  }

  payAccount = () => {
    const options = {
      description: 'Credits towards consultation',
      image: 'https://i.imgur.com/3g7nmJC.png',
      currency: 'INR',
      key: 'rzp_test_9U0bdWxnAUEpRm',
      amount: '5000',
      name: 'ZUL',
      prefill: {
        email: 'sagar.vivek159@gmail.com',
        contact: '9538346231',
        name: 'Razorpay Software'
      },
      theme: { color: '#F37254' }
    };
    RazorpayCheckout.open(options).then((data) => {
      // handle success
      console.log("data", data);
      alert(`Success: ${data.razorpay_payment_id}`);
    }).catch((error) => {
      // handle failure
      console.log("error", error);
      alert(`Error: ${error.code} | ${error.description}`);
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <List>
          <ListItem onPress={this.deleteAccount}>
            <Left>
              <View style={styles.list}>
                <View style={{ alignItems: 'center', alignSelf: 'center', width: 50 }}>
                  <Icon name="user-times" size={20} color="grey" />
                </View>
                <View><Text style={styles.listText}>Delete Account</Text></View>

              </View>
            </Left>

            <Body />
            <Right>
              <Icon name="angle-right" size={20} color="grey" />
            </Right>
          </ListItem>
          {/* <ListItem onPress={this.payAccount}>
            <Left>
              <View style={styles.list}>
                <View style={{ alignItems: 'center', alignSelf: 'center', width: 50 }}>
                  <Icon name="user-times" size={20} color="grey" />
                </View>
                <View><Text style={styles.listText}>Pay Account</Text></View>

              </View>
            </Left>

            <Body />
            <Right>
              <Icon name="angle-right" size={20} color="grey" />
            </Right>
          </ListItem> */}
        </List>
      </View>
    );
  }
}
const menuItemFontStyle = fontMaker({ family: 'Montserrat', weight: 'Regular' });
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white'
  },
  list: {
    flex: 1,
    flexDirection: 'row',
  },
  listText: {
    fontSize: 16,
    color: 'black',
    paddingLeft: 5,
    width: '100%',
    ...menuItemFontStyle
  },
});
export default connect(mapStateToProps, mapDispatchToProps)(AccountSetting);
