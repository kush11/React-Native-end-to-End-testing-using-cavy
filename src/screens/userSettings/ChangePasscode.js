import React, { Component } from 'react';
import {
  Platform, View, StyleSheet, TextInput, Dimensions, TouchableOpacity, ImageBackground
} from 'react-native';
import { Text, Toast } from 'native-base';
import { connect } from 'react-redux';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { SCLAlert, SCLAlertButton } from 'react-native-scl-alert';
import { BASE_URL, headers } from '../../api/config/Config';
import { savePasscodeInfo } from '../../repository/login/LoginRepository';
import { fontMaker, regularButtonFont, defaultModalFont } from '../../components/utility/fonts/FontMaker';

const windowDimensions = Dimensions.get('window');

const mapStateToProps = state => ({
  name: state.User.name,
  mobile: state.User.mobile,
  currentFlow: state.Assessment.currentFlow
});

export class Passcode extends Component {
  state = {
    keyValues: {
      codeOne: '',
      codeTwo: '',
      codeThree: '',
      codeFour: '',
      codeFive: '',
      codeSix: '',
      codeSeven: '',
      codeEight: '',
      codeNine: '',
      codeTen: '',
      codeEleven: '',
      codeTwelve: '',
    },
    buttonDisable: true,
    showSCLAlert: false,
    alertTitle: ''
  }

  curFocInput = 1;

  setDataAndUpdateState = (curFocInputVal, text, firstProp, secondProp) => {
    const { keyValues } = this.state;
    if (!Number.isNaN(text)) {
      keyValues[firstProp] = text;
      if (text !== '' && secondProp !== '' && this.curFocInput !== 4 && this.curFocInput !== 8) {
        keyValues[secondProp] = '';
      }
      this.curFocInput = curFocInputVal;

      if (this.curFocInput > 0 && this.curFocInput !== 4 && this.curFocInput !== 8 && this.curFocInput < 12 && text !== '') {
        this.curFocInput += 1;
      }
      this.refs[this.curFocInput].focus();
    } else {
      keyValues[firstProp] = '';
    }
    this.setState({ keyValues });
    this.buttonDisable();
  }

  isEmptyString = str => str === '';


  buttonDisable = () => {
    const { keyValues } = this.state;
    setTimeout(() => {
      const hasEmpty = Object.values(keyValues)
        .slice(4, 12)
        .some(this.isEmptyString);
      if (!hasEmpty) {
        this.setState({ buttonDisable: false });
      } else {
        this.setState({ buttonDisable: true });
      }
    }, 300);
  }

  handleOpen = (title) => {
    this.setState({ showSCLAlert: true, alertTitle: title });
  }

  handleClose = () => {
    this.setState({ showSCLAlert: false, alertTitle: '' });
  }


  submitPasscode = () => {
    const { keyValues } = this.state;
    const passcodeOne = keyValues.codeOne + keyValues.codeTwo
      + keyValues.codeThree + keyValues.codeFour;
    const passcodeTwo = keyValues.codeFive + keyValues.codeSix
      + keyValues.codeSeven + keyValues.codeEight;
    const passcodeThree = keyValues.codeNine + keyValues.codeTen
      + keyValues.codeEleven + keyValues.codeTwelve;

    const { name, navigation } = this.props;

    if (passcodeThree === passcodeTwo) {
      fetch(`${BASE_URL}/api/changePasscode`, {
        method: 'POST',
        headers,
        body: JSON.stringify({
          uName: name,
          passcode: passcodeOne,
          newPasscode: passcodeTwo
        })
      }).then(response => response.json())
        .then((response) => {
          Platform.OS === 'ios'
            ? Toast.show({
              text: response.message,
              duration: 2000,
              type: 'default',
              position: 'top'
            })
            : Toast.show({
              text: response.message,
              duration: 2000,
              type: 'default'
            });
          setTimeout(() => {
            response.valid === 'true' ? navigation.navigate('UserSetting') : null;
          }, 2000);
          savePasscodeInfo(passcodeTwo);
        })
        .catch(() => { });
    } else {
      this.handleOpen('Oops');
    }
  }

  generateInputs = (inputsData, startIndex) => inputsData.map((input, index) => {
    const { keyValues } = this.state;
    const value = keyValues[input.firstProp];
    return (
      <View style={{ flex: 1 }} key={input.firstProp}>
        <TextInput
          ref={startIndex + index + 1}
          underlineColorAndroid="#000"
          secureTextEntry
          placeholderTextColor="#000"
          keyboardType="numeric"
          value={value}
          style={styles.input}
          maxLength={1}
          onChangeText={
            (text) => {
              this.setDataAndUpdateState(startIndex + index + 1,
                text,
                input.firstProp,
                input.secondProp);
            }
          }
        />
      </View>
    );
  });

  render() {
    const { showSCLAlert, alertTitle, buttonDisable } = this.state;

    const inputsData = [
      { firstProp: 'codeOne', secondProp: 'codeTwo' },
      { firstProp: 'codeTwo', secondProp: 'codeThree' },
      { firstProp: 'codeThree', secondProp: 'codeFour' },
      { firstProp: 'codeFour', secondProp: 'codeFive' },
      { firstProp: 'codeFive', secondProp: 'codeSix' },
      { firstProp: 'codeSix', secondProp: 'codeSeven' },
      { firstProp: 'codeSeven', secondProp: 'codeEight' },
      { firstProp: 'codeEight', secondProp: 'codeNine' },
      { firstProp: 'codeNine', secondProp: 'codeTen' },
      { firstProp: 'codeTen', secondProp: 'codeEleven' },
      { firstProp: 'codeEleven', secondProp: 'codeTwelve' },
      { firstProp: 'codeTwelve', secondProp: '' }
    ];
    const inputsFirstRow = [...inputsData].slice(0, 4);
    const inputsSecondRow = [...inputsData].slice(4, 8);
    const inputsThirdRow = [...inputsData].slice(8, 12);

    return (
      <ImageBackground
        resizeMode="cover"
        style={{
          flex: 1, paddingTop: Platform.OS === 'ios' ? 60 : 0, height: windowDimensions.height, width: windowDimensions.width, position: 'absolute', backgroundColor: 'white', padding: 10
        }}
      >
        <SCLAlert
          theme="danger"
          show={showSCLAlert}
          title={alertTitle}
          subtitle="New Passcode and Confirm Passcode not matched!!!"
          onRequestClose={this.handleClose}
          titleStyle={{ ...defaultModalFont }}
          subtitleStyle={{ ...defaultModalFont }}
        >
          <SCLAlertButton
            theme="danger"
            onPress={this.handleClose}
            textStyle={{ ...regularButtonFont }}
          >
            CLOSE
          </SCLAlertButton>
        </SCLAlert>
        <KeyboardAwareScrollView
          keyboardShouldPersistTaps="handled"
          keyboardDismissMode="interactive"
          enableOnAndroid
        >

          <View>
            <Text style={{
              marginTop: 20, marginLeft: 5, color: 'black', fontSize: 16, ...titleFontStyle
            }}
            >
              Current Passcode
            </Text>
            <Text style={{
              marginLeft: 5, color: '#bbb', fontSize: 14, ...titleFontStyle
            }}
            >
              (Leave blank if never set previously)
            </Text>
            <View style={{ flexDirection: 'row' }}>
              {this.generateInputs(inputsFirstRow, 0)}
            </View>
            <Text style={{
              marginTop: 10, marginLeft: 5, color: 'black', fontSize: 16, ...titleFontStyle
            }}
            >
              New Passcode
            </Text>
            <View style={{ flexDirection: 'row' }}>
              {this.generateInputs(inputsSecondRow, 4)}
            </View>
            <Text style={{
              marginTop: 10, marginLeft: 5, color: 'black', fontSize: 16, ...titleFontStyle
            }}
            >
              Confirm New Passcode
            </Text>
            <View style={{ flexDirection: 'row' }}>
              {this.generateInputs(inputsThirdRow, 8)}
            </View>
            <TouchableOpacity
              style={buttonDisable ? styles.disableBtn : styles.zulBtn}
              disabled={buttonDisable}
              onPress={() => this.submitPasscode()}
            >
              <Text style={styles.whiteText}>{'Submit'.toUpperCase()}</Text>
            </TouchableOpacity>
          </View>
        </KeyboardAwareScrollView>
      </ImageBackground>
    );
  }
}

const titleFontStyle = fontMaker({ family: 'OpenSans', weight: 'Regular' });
const styles = StyleSheet.create({
  loginLogo: {
    marginTop: 60,
    marginBottom: 20
  },
  input: {
    paddingHorizontal: 10,
    paddingVertical: 10,
    fontSize: 20,
    marginHorizontal: 5,
    borderRadius: 5,
    textAlign: 'center',
    color: '#000',
    ...Platform.select({
      ios: {
        borderBottomColor: '#000',
        borderBottomWidth: 1,
      }
    }),
  },
  zulBtn: {
    backgroundColor: '#41ab3e',
    alignItems: 'center',
    marginVertical: 15,
    paddingVertical: 12,
    marginTop: 30,
    borderRadius: 8,
    marginRight: 80,
    marginLeft: 80
  },
  disableBtn: {
    backgroundColor: '#e0e0e0',
    alignItems: 'center',
    marginVertical: 15,
    paddingVertical: 12,
    marginTop: 30,
    borderRadius: 8,
    marginRight: 80,
    marginLeft: 80
  },
  whiteText: {
    color: '#fff',
    ...regularButtonFont
  },
  blackText: { color: '#000' },
  blackMatLayer: {
    backgroundColor: 'white',
    position: 'absolute',
    top: 20,
    bottom: 40,
    left: 10,
    right: 10
  }
});

export default connect(mapStateToProps, null)(Passcode);
