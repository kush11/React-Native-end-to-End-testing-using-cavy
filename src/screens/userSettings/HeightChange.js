import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  TouchableOpacity, PickerIOS
} from 'react-native';
import { connect } from 'react-redux';
import Picker from 'react-native-wheel-picker';
import { updateHeight, updateBmi } from '../../store/actions/index';
import { BASE_URL, headers } from '../../api/config/Config';
import CalculateBmi from '../../components/utility/bmi/Bmi';
import { regularButtonFont, fontMaker } from '../../components/utility/fonts/FontMaker';


const PickerItemIOS = PickerIOS.Item;

const mapStateToProps = state => ({
  name1: state.User.name,
  uHeight: state.User.height,
  uWeight: state.User.weight,
});
const mapDispatchToProps = dispatch => ({
  updatedHeight: height => dispatch(updateHeight(height)),
  updatedBmi: bmi => dispatch(updateBmi(bmi))
});
const PickerItem = Picker.Item;

export class HeightChange extends Component {
  constructor(props) {
    super(props);
    const { uHeight } = this.props;
    this.state = {
      selectedItem: uHeight === '' ? 0 : uHeight,
      itemList: []

    };
    const { itemList } = this.state;
    for (let i = 0; i <= 300; i += 1) {
      itemList[i] = i;
    }
  }

  onPickerSelect(index) {
    this.setState({ selectedItem: index });
  }

  onAddItem = () => {
    const {
      name1, uWeight, navigation, updatedHeight, updatedBmi
    } = this.props;
    const { itemList, selectedItem } = this.state;
    fetch(`${BASE_URL}/api/userData`, {
      method: 'POST',
      headers,
      body: JSON.stringify({
        name: name1,
        height: itemList[selectedItem]
      })
    }).then(response => response.json(),
      updatedHeight(itemList[selectedItem]))
      .then(updatedBmi(CalculateBmi(itemList[selectedItem], uWeight)))
      .then(navigation.navigate('PersonalSetting'))
      .catch(() => { });
  }

  render() {
    const { selectedItem, itemList } = this.state;
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Please select
        </Text>

        {Platform.OS === 'ios'
          ? (
            <PickerIOS
              style={{
                marginTop: 20, alignSelf: 'center', width: '100%', height: 180, backgroundColor: '#dcdcdc',
              }}
              selectedValue={selectedItem}
              itemStyle={{ color: 'black', fontSize: 26 }}
              onValueChange={index => this.onPickerSelect(index)}
            >
              {itemList.map((value, i) => (
                <PickerItemIOS
                  key={`number${value.toString()}`}
                  value={i}
                  label={value.toString()}
                />
              ))}
            </PickerIOS>
          )
          : (
            <Picker
              style={{
                width: '100%',
                height: 180,
                justifyContent: 'center',
                alignItems: 'center',
                backgroundColor: '#eee',
                marginTop: 20
              }}
              selectedValue={selectedItem}
              itemStyle={{ color: 'black', fontSize: 26 }}
              onValueChange={index => this.onPickerSelect(index)}
            >
              {itemList.map((value, i) => (
                <PickerItem label={value.toString()} value={i} key={`number${value.toString()}`} />
              ))}
            </Picker>
          )}
        <Text style={{
          margin: 20,
          color: '#000000',
          fontSize: 20,
          justifyContent: 'center',
          marginTop: 30,
          alignItems: 'center',
          alignSelf: 'center',
          ...titleFontStyle
        }}
        >
          Selected :
          {' '}
          {itemList[selectedItem]}
          {' '}
          cm
        </Text>
        <TouchableOpacity style={styles.setBmiBtn} onPress={this.onAddItem}>
          <Text style={styles.textWhite}>{'Set Height'.toUpperCase()}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const titleFontStyle = fontMaker({ family: 'Montserrat', weight: 'Regular' });
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white'
  },
  setBmiBtn: {
    backgroundColor: '#41ab3e',
    alignItems: 'center',
    marginVertical: 15,
    paddingVertical: 12,
    borderRadius: 8,
    marginRight: 60,
    marginLeft: 60

  },
  textWhite: {
    color: '#ffffff',
    fontSize: 15,
    ...regularButtonFont
  },
  welcome: {
    fontSize: 16,
    textAlign: 'center',
    marginTop: 15,
    color: 'black',
    ...titleFontStyle
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(HeightChange);
