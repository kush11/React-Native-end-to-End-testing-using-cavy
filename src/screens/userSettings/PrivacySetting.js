import React, { Component } from 'react';
import {
  Platform, View, StyleSheet, Text, Switch, Alert
} from 'react-native';
import {
  Left, Body, Right, List, ListItem, Toast
} from 'native-base';
import { saveTouchIdInfo, isTouchIdEnabled } from '../../repository/login/LoginRepository';
import { fontMaker } from '../../components/utility/fonts/FontMaker';

export class PrivateSetting extends Component {
  state = { isRegister: false };

  componentDidMount() {
    isTouchIdEnabled().then((isTouch) => {
      this.setState({ isRegister: isTouch });
    });
  }

  toastMessage = () => {
    Toast.show({
      text: 'This feature will be Available Soon',
      duration: 2000,
      type: 'default'
    });
  }

  changeAuthMode = () => {
    const { isRegister } = this.state;
    const userinfo = {};
    if (!isRegister) {
      Alert.alert(
        'Enable Touch Id',
        'Do you want to enable touch id?',
        [
          {
            text: 'NO',
            onPress: () => {
              userinfo.enableTouchId = 'false';
              saveTouchIdInfo(userinfo);
              this.setState({ isRegister: false });
            },
            style: 'cancel'
          },
          {
            text: 'YES',
            onPress: () => {
              userinfo.enableTouchId = 'true';
              saveTouchIdInfo(userinfo);
              this.setState({ isRegister: true });
            }
          },
        ],
        { cancelable: false }
      );
    } else {
      Alert.alert(
        'Disable Touch Id',
        'Do you want to disable touch id?',
        [
          {
            text: 'NO',
            onPress: () => {
              userinfo.enableTouchId = 'true';
              saveTouchIdInfo(userinfo);
              this.setState({ isRegister: true });
            },
            style: 'cancel'
          },
          {
            text: 'YES',
            onPress: () => {
              userinfo.enableTouchId = 'false';
              saveTouchIdInfo(userinfo);
              this.setState({ isRegister: false });
            }
          },
        ],
        { cancelable: false }
      );
    }
  }

  render() {
    const { isRegister } = this.state;
    const authModeSelectionSection = (
      <View style={styles.authModeContainer}>
        {/* <View ><Text style={styles.authModeText}>OFF</Text></View> */}
        {(Platform.OS === 'ios')
          ? (
            <Switch
              onValueChange={this.changeAuthMode}
              thumbTintColor={isRegister ? '#4b7aa5' : '#cccccc'}
              onTintColor="#00264a"
              value={isRegister}
              style={{ transform: [{ scaleX: 1 }, { scaleY: 1 }] }}
            />
          )
          : (
            <Switch
              thumbTintColor={isRegister ? '#4b7aa5' : '#cccccc'}
              onTintColor="#00264a"
              value={isRegister}
              style={{ transform: [{ scaleX: 1.3 }, { scaleY: 1.3 }] }}
              onValueChange={this.changeAuthMode}
            />
          )}

      </View>
    );

    return (
      <View style={styles.container}>
        <List>
          <ListItem>

            <Left>
              <View style={styles.list}>
                <View><Text style={styles.listText}>Enable Touch Id</Text></View>

              </View>
            </Left>


            <Body />
            <Right style={{ alignContent: 'center' }}>
              <View>
                {authModeSelectionSection}
              </View>

            </Right>
          </ListItem>
        </List>
      </View>
    );
  }
}

const itemFontStyle = fontMaker({ family: 'OpenSans', weight: 'Regular' });
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white'
  },
  authModeContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly'
  },
  authModeText: {
    color: 'black',
    fontWeight: 'bold',
    margin: 10,
  },
  list: { flexDirection: 'row' },
  listText: {
    fontSize: 18,
    textAlign: 'center',
    color: 'black',
    ...itemFontStyle
  },
});
export default PrivateSetting;
