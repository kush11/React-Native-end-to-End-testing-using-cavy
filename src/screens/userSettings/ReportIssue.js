import React, { Component } from 'react';
import {
  View, StyleSheet, TouchableOpacity, Linking, TextInput
} from 'react-native';
import { Text } from 'native-base';
// import ImagePicker from 'react-native-image-picker';
import { SCLAlert, SCLAlertButton } from 'react-native-scl-alert';
import { regularButtonFont, defaultModalFont } from '../../components/utility/fonts/FontMaker';

export class Register extends Component {
  state = {
    query: '',
    // pickedImage1: '',
    // pickedImage2: '',
    // pickedImage3: '',
    // pickedImage1Path: '',
    // pickedImage2Path: '',
    // pickedImage3Path: '',
    // count: 1,
    // spinner: false,
    showSCLAlert: false
  };

  // pickImageHandler = (choice) => {
  //   ImagePicker.showImagePicker(
  //     {
  //       title: 'Choose Profile Image'
  //     },
  //     (response) => {
  //       if (response.didCancel) {
  //         console.log('User cancelled image picker');
  //       } else if (response.error) {
  //         console.log('ImagePicker Error: ', response.error);
  //       } else {
  //         console.log('respnse data', response);
  //         const source = {
  //           uri: response.uri,
  //           // base64: response.data
  //         };
  //         if (choice === 1) {
  //           this.setState({
  //             pickedImage1: source,
  //             count: 1,
  //             pickedImage1Path: response.path
  //             // spinner: true
  //           });
  //         } else if (choice === 2) {
  //           this.setState({
  //             pickedImage2: source,
  //             count: 1,
  //             pickedImage2Path: response.path
  //             // spinner: true
  //           });
  //         } else {
  //           this.setState({
  //             pickedImage3: source,
  //             count: 1,
  //             pickedImage3Path: response.path
  //             // spinner: true
  //           });
  //         }
  //       }
  //     }
  //   );
  // };

  handleEmail = () => {
    const { query } = this.state;
    const { navigation } = this.props;
    Linking.openURL(`mailto:support@zinguplife.zohodesk.com?subject=Question about Zing up life&body=${query}`);
    navigation.goBack();
  }

  handleClose = () => {
    this.setState({ showSCLAlert: false });
  }

  handleOpen = () => {
    this.setState({ showSCLAlert: true });
  }

  render() {
    const { showSCLAlert, query } = this.state;
    return (
      <View style={styles.container}>
        <SCLAlert
          theme="danger"
          show={showSCLAlert}
          title="Oops!"
          subtitle="Query field must be filled"
          onRequestClose={this.handleClose}
          titleStyle={{ ...defaultModalFont }}
          subtitleStyle={{ ...defaultModalFont }}
        >
          <SCLAlertButton
            theme="danger"
            onPress={this.handleClose}
            textStyle={{ ...regularButtonFont }}
          >
            OK
          </SCLAlertButton>
        </SCLAlert>
        <View style={{ marginLeft: 25, marginRight: 25 }}>
          <TextInput
            underlineColorAndroid="#144E76"
            maxLength={500}
            multiline
            placeholder="Please describe your problem"
            style={styles.customStyle}
            onChangeText={(text) => { this.setState({ query: text }); }}
            accessible
            accessibilityLabel="Enter Query"
            accessibilityHint="Provide Query"
          />
          <TouchableOpacity
            style={styles.zulBtn}
            onPress={() => (query.trim(' ').length !== 0 ? this.handleEmail() : this.handleOpen())}
          >
            <Text style={styles.whiteText}>NEXT</Text>
          </TouchableOpacity>
        </View>

      </View>
    );
  }
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white'
  },
  blackMatLayer: {
    backgroundColor: '#00000054',
    position: 'absolute',
    top: 20,
    bottom: 40,
    left: 10,
    right: 10
  },
  customStyle: {
    fontSize: 20,
    borderBottomLeftRadius: 10,
    borderBottomRightRadius: 10,
    borderColor: '#41ab3e',
    height: 100,
    width: '100%'
  },
  zulBtn: {
    backgroundColor: '#41ab3e',
    alignItems: 'center',
    marginVertical: 15,
    paddingVertical: 12,
    marginTop: 30,
    borderRadius: 8,
    marginRight: 60,
    marginLeft: 60

  },
  whiteText: {
    color: '#fff',
    fontSize: 15,
    // fontWeight: 'bold',
    ...regularButtonFont
  },
  image: {
    borderWidth: 1,
    borderColor: 'black',
    alignItems: 'center',
    justifyContent: 'center',
    height: 200,
    width: 200,
    borderRadius: 25
  },
  imagesrc: {
    height: 200,
    width: 200,
    borderRadius: 25
  }
});

export default Register;
