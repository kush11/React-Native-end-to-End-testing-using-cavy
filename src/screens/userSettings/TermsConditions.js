import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { fontMaker } from '../../components/utility/fonts/FontMaker';

const termsConditions = (props) => {
  const { children } = props;
  return (
    <View>
      <Text style={styles.listText}>
        {children}
      </Text>
    </View>
  );
};

const termsFontStyle = fontMaker({ family: 'OpenSans', weight: 'Regular' });
const styles = StyleSheet.create({
  container: { flex: 1 },
  list: { flexDirection: 'row' },
  listText: {
    fontSize: 18,
    textAlign: 'center',
    color: 'black',
    ...termsFontStyle
  },
});
export default termsConditions;
