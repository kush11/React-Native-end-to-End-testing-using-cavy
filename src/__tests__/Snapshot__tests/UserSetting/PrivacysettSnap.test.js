import 'react-native';
import 'isomorphic-fetch';
import React from 'react';
import Renderer from 'react-test-renderer';
import { PrivateSetting } from '../../../screens/userSettings/PrivacySetting';

it('snapshots', () => {
  const tree = Renderer.create(<PrivateSetting />).toJSON();
  expect(tree).toMatchSnapshot();
});
