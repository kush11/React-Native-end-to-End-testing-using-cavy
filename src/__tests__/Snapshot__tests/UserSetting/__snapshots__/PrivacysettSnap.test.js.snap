// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    closeOnRowBeginSwipe={false}
    closeOnRowPress={true}
    closeOnScroll={true}
    directionalDistanceChangeThreshold={2}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    leftOpenValue={0}
    previewFirstRow={false}
    recalculateHiddenLayout={false}
    rightOpenValue={0}
    style={Object {}}
    swipeToOpenPercent={50}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 16,
            "paddingBottom": 13,
            "paddingRight": 16,
            "paddingTop": 13,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "OpenSans-Regular",
                    "fontSize": 18,
                    "textAlign": "center",
                  }
                }
              >
                Enable Touch Id
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": null,
              "alignSelf": null,
              "flex": 1,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "alignSelf": "center",
                "flex": 0.28,
                "paddingBottom": null,
                "paddingLeft": null,
                "paddingRight": null,
                "paddingTop": null,
              },
              Object {
                "alignContent": "center",
              },
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            >
              <RCTSwitch
                disabled={false}
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onTintColor="#00264a"
                onValueChange={[Function]}
                style={
                  Array [
                    Object {
                      "height": 31,
                      "width": 51,
                    },
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                    },
                  ]
                }
                thumbTintColor="#cccccc"
                value={false}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
